google/cloud/firestore_v1/vector.py

```diff
@@ -13,19 +13,32 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import List, Sequence
+import collections
 
+from typing import Tuple, Sequence
 
-class Vector:
+class Vector (collections.abc.Sequence):
     r"""A class to represent Firestore Vector in python.
 
     Underline it'll be converted to a map representation in Firestore API.
     """
 
-    _value: List[float] = []
+    _value: Tuple[float] = ()
 
     def __init__(self, value: Sequence[float]):
-        self._value = [float(v) for v in value]
+        self._value = tuple([float(v) for v in value])
+
+    def __getitem__(self, index: int):
```

Comment:
__getitem__ can also be called with a slice

But I'd suggest keeping things simple and just delegating everything to the inner tuple, no need to do any processing here: 

```
def __getitem__(self, arg):
  return self._value[arg]
```

Reply:
Done.